Extension/src/LanguageServer/references.ts

```diff
@@ -343,7 +344,20 @@ export class ReferencesManager {
             this.referencesChannel.show(true);
         }
 
-        if (this.client.ReferencesCommandMode === ReferencesCommandMode.Rename) {
+        let currentReferenceCommandMode: ReferencesCommandMode = this.client.ReferencesCommandMode;
+        if (referencesResult.isFinished) {
+            this.symbolSearchInProgress = false;
+            clearInterval(this.referencesDelayProgress);
+            if (this.currentUpdateProgressTimer) {
+                clearInterval(this.currentUpdateProgressTimer);
+                this.currentUpdateProgressResolve();
+                this.currentUpdateProgressResolve = null;
+                this.currentUpdateProgressTimer = null;
+            }
+            this.client.setReferencesCommandMode(ReferencesCommandMode.None);
+        }
+
+        if (currentReferenceCommandMode === ReferencesCommandMode.Rename) {

```

Comment:
Can this be an "else if"?

Reply:
Looks like it can't...

Reply:
Are you suggesting if isFinished, don't do the rest?  I think the rest is still needed even if isFinished.